pkg/catalog/server.go

```diff
@@ -90,16 +92,21 @@ func (p *Generator) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 		if err != nil {
 			p.errs = append(p.errs, err)
 		}
+		p.GeneratedFilesMutex.Lock()
```

Comment:
nit: Could potentially use `sync.RWMutex` above and then use `p.GeneratedFilesMutex.RLock()` here, since we're okay to read the map concurrently